### PR TITLE
Add more unit tests for UdpFramed

### PR DESCRIPTION
**Pull request by kpp**
For https://github.com/tokio-rs/tokio/pull/1517 and https://github.com/tokio-rs/tokio/issues/1510

### CODE REVIEW

File: tokio-udp/tests/udp.rs [hunk 1]
```diff
@@ -315,6 +315,75 @@ fn send_framed_lines_codec() {
...
+        let a = t!(send.wait());
+
+        let msg = b"1\r\n".to_vec(); // fits ok
+        let send = a.send((msg.clone(), b_addr));
+        t!(send.wait());
+
+        let mut b = Stream::wait(b);
+
+        let hello_world = b.next().unwrap();
+        dbg!(&hello_world);
```
Comment (LucioFranco) [inline tokio-udp/tests/udp.rs]: do we need this?

File: tokio-udp/tests/udp.rs [hunk 2]
```diff
@@ -315,6 +315,75 @@ fn send_framed_lines_codec() {
     assert_eq!(recv.next(), Some(("3".to_string(), a_addr)));
 }
 
+#[test]
+fn recv_framed_codec_errs() {
+    drop(env_logger::try_init());
```
Comment (LucioFranco) [inline tokio-udp/tests/udp.rs]: do we also need this?

Comment (LucioFranco) [reply tokio-udp/tests/udp.rs]: meh I see we have this in other tests.

File: tokio-udp/tests/udp.rs (hunk 1 above)
Comment (LucioFranco) [inline tokio-udp/tests/udp.rs]:
```suggestion
```

**Comment by LucioFranco** [comment]
@kpp can you resolve that one dbg! then I can merge and publish, thanks for this!

File: tokio-udp/tests/udp.rs (hunk 1 above)
Comment (kpp) [reply tokio-udp/tests/udp.rs]: oops

**Pull request merged**